### PR TITLE
Add Arch / Manjaro section for Linux setup instructions

### PR DESCRIPTION
**Pull request by riblee**
## Description

Add Arch / Manjaro section for Linux setup instructions

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* #10492 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by Erovia** [comment]
I'm not entirely on board with this in its current form.
While I think it's great to have the CLI packaged for different distros, I need to be made clear that it's done by "3rd-parties" and not by QMK.
Something similar that we currently have with the AUR package.

**Comment by tzarc** [comment]
@Erovia are you happy with its current form?
@riblee we'll need to get conflicts sorted, but if @Erovia is happy we should be right to get this in.

**Comment by fauxpark** [comment]
The `qmk` package actually marks a bunch of stuff as optional that really shouldn't be, so I don't think we should be recommending it until that is fixed.

https://www.archlinux.org/packages/community/any/qmk/

**Comment by Erovia** [comment]
Github does not allow me to give an accurate suggestion, because commenting on removed lines is not possible, so please bear with me.

I'd prefer if the current instructions for distros were kept, especially now that fauxpark added a few other distros to the list.
Below that, we could have a "Community packages" list, which could have your changes for Arch.
So having the official (from QMK project's point of view) on the top, below the distro-specific stuff.
I'm sure users will find it either way, because we already have ppl coming for help, because they installed the Arch package, but missed a few "optdepends".

I did some testing now, as I've noticed some activity on the Arch repo and as of the writing of this reply, the following additional packages are **needed** for compilation:
- make
- diffutils
- arm-none-eabi-newlib (Arm-only)

This information needs to be visible as some users might not be knowledgeable enough to know what to look for or just forget to check the optional dependencies.

Sorry for the ping, @Foxboron, but I think you should weigh in on this last part.

**Comment by Foxboron** [comment]
Yo,

Since qmk is in practice just a tool to help run make for different keyboards. The tradeoff is between having all the possible tools people *might* use always installed, even if they just need half of it, or keep everything together (people complaining about "bloat"). In practice I don't really care, storage is cheap. So whatever the community think is correct.

`make` is part of `base-devel` which is expected on Arch if you do any kind of compilation (so not going to be a dependency). `diffutils` can be added but I haven't figured out completely in which cases it's used. `arm-none-eabi-newlib` can also be added, I might have missed moving this one as a dependency during the last ping on Discord.

However, I object to listing it as "Arch/Manjaro". This is packaged for Arch, and I can't provide any guarantees for what Manjaro does downstream.

Does this clear up stuff or is there anything I'm missing?

For the sake of reference, the upstream `PKGBUILD` is in my monorepo:
https://github.com/Foxboron/archlinux-pkgbuilds/blob/master/qmk/PKGBUILD

**Comment by fauxpark** [comment]
`cmp` appears to be used by `tmk_core/rules.mk`:
https://github.com/qmk/qmk_firmware/blob/3a69d688f2d9bc18a1e8a1f1951025d107c52855/tmk_core/rules.mk#L353-L372

**Review by zvecr — CHANGES_REQUESTED** [review]
Needs a rebase due to recent changes. I think it could also be better to split into something like "manual" and "community", as to put more emphasis on the `it's maintained by a community member` line.

**Comment by riblee** [comment]
Sorry for the delay, I've added a `Community packages` section to the docs with the packages in the official repositories and AUR.

I could add a block quote with on the `it's maintained by a community member` line like this:
> **NOTE**: These packages are maintained by a community member

File: docs/newbs_getting_started.md [hunk 1]
```diff
@@ -93,7 +93,15 @@ Install the QMK CLI by running:
...
-On Arch-based distros you can also try the `qmk` package from AUR (**NOTE**: this package is maintained by a community member, and at the time of writing marks some dependencies as optional that should not be):
+#### Community packages
+
+**NOTE**: These packages are maintained by a community member
+
+On Arch-based distros you can install the CLI from the official repositories:
+
+    sudo pacman -S qmk
+
+You can also try the `qmk` package from AUR (**NOTE**: this package at the time of writing marks some dependencies as optional that should not be):
```
Comment (fauxpark) [inline docs/newbs_getting_started.md:104]:
```suggestion
On Arch-based distros you can install the CLI from the official repositories (NOTE: at the time of writing this package marks some dependencies as optional that should not be):

    sudo pacman -S qmk

You can also try the `qmk` package from AUR:
```
Looks like it is actually the "official" package that has the dependency issues.

File: docs/newbs_getting_started.md [hunk 2]
```diff
@@ -93,7 +93,15 @@ Install the QMK CLI by running:
 
     python3 -m pip install --user qmk
 
-On Arch-based distros you can also try the `qmk` package from AUR (**NOTE**: this package is maintained by a community member, and at the time of writing marks some dependencies as optional that should not be):
+#### Community packages
+
+**NOTE**: These packages are maintained by a community member
```
Comment (fauxpark) [inline docs/newbs_getting_started.md:98]:
```suggestion
#### Community Packages

These packages are maintained by community members, so may not be up to date or completely functional. If you encounter problems, please report them to their respective maintainers.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by Erovia — APPROVED** [review]
Yeah, this looks good to me.

**Pull request merged**